### PR TITLE
Upgrade walli grpc getInfo() 

### LINKED ISSUE #45: Proper getInfo command
Right now the `getInfo` in #39 is not showing meaningful information to the user.

What it should show is:
- what network(s) the nodes of walli-server are connected to (mainnet, testnet)
- the output of the `getInfo` call of XUD (at least the parts that are not redundant to what walli-server is showing by its own) 
- whether the on-chain nodes (just BTCD right now) are in sync and other *important* data about them
- channel counts, balances and connectivity information about the off-chain (LND) nodes

Feel free to suggest adding more things to the command @dopetard @ImmanuelSegol

### PR DESCRIPTION
**Pull request by ImmanuelSegol**
Closes #45 
From where should I get the `walli-server`  version, I simply hard set it for the time being.

### CODE REVIEW

**Comment by michael1011** [comment]
@ImmanuelSegol you can get the version from `package.json` with:
```
const version: string = require('../package.json').version;
```

File: lib/chain/BtcdClient.ts [hunk 1]
```diff
@@ -69,4 +69,4 @@ class BtcdClient extends BaseClientClass implements ChainClient, BtcdClient {
 }
 
 export default BtcdClient;
-export { BtcdConfig };
+export { BtcdConfig, Info as BtcdInfo };
```
Comment (michael1011) [inline lib/chain/BtcdClient.ts]: I think the `as` should be used when importing not exporting types

File: lib/lightning/LndClient.ts [hunk 2]
```diff
@@ -161,4 +197,4 @@ class LndClient extends BaseClientClass implements LightningClient {
 }
 
 export default LndClient;
-export { LndConfig };
+export { LndConfig, Info as LndInfo };
```
Comment (michael1011) [inline lib/lightning/LndClient.ts]: Same here

File: lib/service/Service.ts [hunk 3]
```diff
@@ -1,11 +1,27 @@
...
 class Service{
-  constructor(private btcdClient: BtcdClient) {}
 
-  public getInfo = async (): Promise<object> => {
-    const info = await this.btcdClient.getInfo();
-    return info;
+  constructor(private logger: Logger, private btcdClient: BtcdClient, private lndClient: LndClient) {}
+
+  public getInfo = async (): Promise<walliInfo> => {
+    const version = require('../../package.json').version;
```
Comment (michael1011) [inline lib/service/Service.ts]: This should be moved outside of the function and class so that `package.json` is only read once (when importing the file) instead of every time `getInfo` is called

File: proto/wallirpc.proto [hunk 4]
```diff
@@ -10,13 +10,34 @@ service Walli {
...
-   bool testnet = 8;
-   float relayfee = 9;
+  string version = 1;
+  BtcdInfo btcdInfo = 2;
+  LndInfo lndInfo = 3;
+}
+
+message LndInfo {
+  string version = 1;
+  repeated string chains = 2;
```
Comment (michael1011) [inline proto/wallirpc.proto]: I think we should just show as little information as needed to be sure the nodes are working as intended because we will have to show that information for every chain and client we want to support. The amount of:
- blocks (for LND and BTCD to make sure LND is in sync with BTCD)
- channels
- peers (which are called `connections` in BTCD)
- and the version of the node 

should be enough but feel free to disagree and discuss other approaches.

File: lib/service/Service.ts [hunk 5]
```diff
@@ -1,11 +1,29 @@
-import BtcdClient from '../chain/BtcdClient';
+import Logger from '../Logger';
+import BtcdClient, { Info as BtcdInfo } from '../chain/BtcdClient';
+import LndClient, { Info as LndInfo } from '../lightning/LndClient';
+
+const packageJson = require('../../package.json');
+
+type walliInfo = {
```
Comment (michael1011) [inline lib/service/Service.ts]: Types should start with an uppercase letter: `WalliInfo`

File: package.json [hunk 6]
```diff
@@ -31,7 +31,7 @@
...
       "darwin": "./node_modules/grpc-tools/bin/protoc --js_out='import_style=commonjs,binary:lib/proto' --ts_out='lib/proto' --grpc_out='lib/proto' --plugin='protoc-gen-grpc=node_modules/.bin/grpc_tools_node_protoc_plugin' --plugin='protoc-gen-ts=node_modules/grpc_tools_node_protoc_ts/bin/protoc-gen-ts' -I='proto' proto/*.proto proto/google/api/*.proto proto/google/protobuf/*.proto",
-      "win32": "node_modules\\grpc-tools\\bin\\protoc --js_out=\"import_style=commonjs,binary:lib\\proto\" --ts_out=\"lib\\proto\" --grpc_out=\"lib\\proto\" --plugin=\"protoc-gen-grpc=node_modules\\.bin\\grpc_tools_node_protoc_plugin.cmd\" --plugin=\"protoc-gen-ts=node_modules\\.bin\\protoc-gen-ts.cmd\" -I=\"proto\" proto\\lndrpc.proto proto\\wallirpc.proto proto\\google\\api\\annotations.proto proto\\google\\api\\http.proto proto\\google\\protobuf\\descriptor.proto"
```
Comment (michael1011) [inline package.json]: Why did you remove the compilation of the LND proto?

File: lib/grpc/GrpcService.ts [hunk 7]
```diff
@@ -17,15 +19,40 @@ class GrpcService {
...
+
+        if (lndInfo.channels) {
+          const channels = new wallirpc.LndChannels();
+          channels.setActive(lndInfo.channels.active);
+          channels.setPending(lndInfo.channels.pending);
+          lndInfo.channels.inactive ? channels.setInactive(lndInfo.channels.inactive) : undefined;
+          lnd.setLndchannels(channels);
+        }
+
+        version ? lnd.setVersion(version) : undefined;
```
Comment (michael1011) [inline lib/grpc/GrpcService.ts]: Why not just `lnd.setVersion(version)`? Is there an error when setting the version to `undefined`?

Comment (ImmanuelSegol) [reply lib/grpc/GrpcService.ts]: yes, there is `lnd.setVersion(version)` cant accept `undefined` .

File: package.json (hunk 6 above)
Comment (ImmanuelSegol) [reply package.json]: Wow didn't notice that, fixed it!

File: package.json [hunk 8]
```diff
@@ -31,7 +31,7 @@
...
+      "win32": "node_modules\\grpc-tools\\bin\\protoc --js_out=\"import_style=commonjs,binary:lib\\proto\" --ts_out=\"lib\\proto\" --grpc_out=\"lib\\proto\" --plugin=\"protoc-gen-grpc=node_modules\\.bin\\grpc_tools_node_protoc_plugin.cmd\" --plugin=\"protoc-gen-ts=node_modules\\.bin\\protoc-gen-ts.cmd\" -I=\"proto\" proto\\lndrpc.proto proto\\wallirpc.proto proto\\google\\api\\annotations.proto proto\\google\\api\\http.proto proto\\google\\protobuf\\descriptor.proto"
```
Comment (michael1011) [inline package.json]: There is an unnecessary space at the end of the line
